lib/_http_outgoing.js

```diff
@@ -90,11 +90,12 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
-    if (typeof callback !== 'function')
-      throw new TypeError('callback must be a function');
+  if (typeof callback === 'function') {
     this.on('timeout', callback);
   }
+  else if (callback) {
+    throw new TypeError('callback must be a function');
```

Comment:
I'm for it *if* there is the `typeof` check. I think @thefourtheye is saying that you can discard the `typeof` check *and* the `else` block and just pass blindly to `.on()` because `.on()` will through a type error if the callback is not a function. At least, that's my understanding. So I'm for *either* approach, really. Don't make no neither nor to me.